core/basicauth/interfaces/authservice.go

```diff
@@ -0,0 +1,74 @@
+package interfaces
+
+import (
+	"context"
+	"errors"
+	"flamingo.me/flamingo/v3/core/auth/domain"
+	"flamingo.me/flamingo/v3/framework/config"
+	"flamingo.me/flamingo/v3/framework/web"
+	"net/http"
+	"net/url"
+)
+
+type (
+	//Authservice - basic auth authservice impplementation
+	Authservice struct {
+		responder *web.Responder
+		realm string
+	}
+
+	//Idendity - is a http basic auth Idendity using plantext Username and Password
+	Idendity struct {
+		Username string
+		Password string
+		role string
+	}
+)
+
+var (
+	_ domain.Idendity = &Idendity{}
+)
+
+// Inject for Authservice
+func (o *Authservice) Inject(responder *web.Responder, config *struct {
+	Users config.Slice `inject:"config:basicauth.users,optional"`
```

Comment:
Maybe users should be provided via something like a UserProvider? So the credentials do not have to be in the config?